lib/engine_wrapper.py

```diff
@@ -1222,25 +1225,29 @@ def get_gaviota(board: chess.Board, game: model.Game,
             or chess.popcount(board.occupied) > gaviota_cfg.max_pieces
             or board.uci_variant != "chess"):
         return None, -3
+
     move: Union[chess.Move, list[chess.Move]]
     move_quality = gaviota_cfg.move_quality
+
     # Since gaviota TBs use dtm and not dtz, we have to put a limit where after it the position are considered to have
     # a syzygy wdl=1/-1, so the positions are draws under the 50 move rule. We use min_dtm_to_consider_as_wdl_1 as a
     # second limit, because if a position has 5 pieces and dtm=110 it may take 98 half-moves, to go down to 4 pieces and
     # another 12 to mate, so this position has a syzygy wdl=2/-2. To be safe, the first limit is 100 moves, which
     # guarantees that all moves have a syzygy wdl=2/-2. Setting min_dtm_to_consider_as_wdl_1 to 100 will disable it
     # because dtm >= dtz, so if abs(dtm) < 100 => abs(dtz) < 100, so wdl=2/-2.
     min_dtm_to_consider_as_wdl_1 = gaviota_cfg.min_dtm_to_consider_as_wdl_1
+
     with chess.gaviota.open_tablebase(gaviota_cfg.paths[0]) as tablebase:
         for path in gaviota_cfg.paths[1:]:
             tablebase.add_directory(path)
 
         try:
             moves = score_gaviota_moves(board, dtm_scorer, tablebase)
 
-            best_wdl = max(map(dtm_to_gaviota_wdl, moves.values()))
-            good_moves = [(move, dtm) for move, dtm in moves.items() if dtm_to_gaviota_wdl(dtm) == best_wdl]
-            best_dtm = min([dtm for move, dtm in good_moves])
+            converted_moves = {move: dtm_to_gaviota_wdl(dtm) for move, dtm in moves.items()}
+            best_wdl = max(converted_moves.values())
+            good_moves = [(move, dtm) for move, dtm in converted_moves.items() if dtm == best_wdl]
```

Comment:
Same problem here. good_moves contains pairs of (move, estimated wdl) instead of (move, dtm).